### DISCUSSION TITLE
Possible integration of Surface Mesh Skeletonization of CGAL library

### DISCUSSION THREAD

**Discussion by polemieux** [cnr-isti-vclab/PyMeshLab#402] [Ideas]
Hi,

I would like to know if anyone is working to integrate the **[surface mesh skeletonization](https://doc.cgal.org/latest/Surface_mesh_skeletonization/index.html#title7)** of the CGAL library. This method seems really useful to segment objects and I would like to test it intensively with python scripts on many shapes. 

If no integration into PyMeshLab is planned, does anyone know how to use the CGAL version through Python or C++? Maybe some GIT repos? We found many repos that had the skeletonization (e.g. [romicgal](https://github.com/romi/romicgal), [r03ert0](https://github.com/r03ert0/skeleton), [starlab-mcfskel]([starlab-mcfskel](https://github.com/taiya/starlab-mcfskel))) but we always ran into compilation errors with the Cmake thing. If anyone has a successfull recipe to make it work with some repo either in a Python or C++ environment, I would appreciate if you can share it. 

Thank you very much.

Pierre

**Comment by alemuntoni** [comment]
I think @Yurand2000 is working on this (among with several other more specific features) on [his fork](https://github.com/Yurand2000/meshlab/tree/devel), that will be merged soon on meshlab, and then will be automatically added also in pymeshlab.

**Comment by polemieux** [comment]
Hi Alessandro,

Thank you for the quick reply. Is it possible to test the code while it is still being integrated or we have to wait until it is released?

Pierre

**Reply by jmespadero** [reply]
Yes, It is possible to test the code. But you will have to compile it by yourself using the build instructions.

**Reply by cignoni** [reply]
Actually if you are logged in github you should be able to access the build generated by github actions
https://github.com/Yurand2000/meshlab/actions/runs/11665285871
and test it directly 
_(edit: just for meshlab, while for pymeshlab you have to build them)_

**Reply by polemieux** [reply]
Hi,

I tried the skeletonization functionnality with Meshlab and it works well. Now I need to run it on several objects. I am using python so I need to access it with pymeshlab. When do you plan to integrate this branch in the main branch?

Also, I saw here (https://github.com/cnr-isti-vclab/PyMeshLab/blob/main/src/README.md) that we can build PyMeshLab from the GitHub. Thus, instead of waiting for Yurand2000 branch to be added to MeshLab to access it through PyMeshLab, is there a faster way to achieve the same on my side with his branch and pybind11 ? 

Thank you very much.

Pierre
